x-pack/legacy/plugins/siem/public/components/ml/tables/convert_anomalies_to_hosts.ts

```diff
@@ -4,24 +4,39 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { Anomalies, AnomaliesByHost } from '../types';
+import { Anomalies, AnomaliesByHost, Anomaly } from '../types';
 import { getHostNameFromInfluencers } from '../influencers/get_host_name_from_influencers';
 
-export const convertAnomaliesToHosts = (anomalies: Anomalies | null): AnomaliesByHost[] => {
+export const convertAnomaliesToHosts = (
+  anomalies: Anomalies | null,
+  hostName?: string
+): AnomaliesByHost[] => {
   if (anomalies == null) {
     return [];
   } else {
     return anomalies.anomalies.reduce<AnomaliesByHost[]>((accum, item) => {
-      if (item.entityName === 'host.name') {
+      if (getHostNameFromEntity(item, hostName)) {
         return [...accum, { hostName: item.entityValue, anomaly: item }];
       } else {
-        const hostName = getHostNameFromInfluencers(item.influencers);
-        if (hostName != null) {
-          return [...accum, { hostName, anomaly: item }];
+        const hostNameFromInfluencers = getHostNameFromInfluencers(item.influencers, hostName);
+        if (hostNameFromInfluencers != null) {
+          return [...accum, { hostName: hostNameFromInfluencers, anomaly: item }];
         } else {
           return accum;
         }
       }
     }, []);
   }
 };
+
+export const getHostNameFromEntity = (anomaly: Anomaly, hostName?: string): boolean => {
```

Comment:
nit - I am just wondering if something like that is more readable
```
if (anomaly.entityName !== 'host.name') {
  return false;
}
if (hostName == null) {
  return true;
}
return anomaly.entityValue === hostName;
```

Reply:
Yes, that is more readable. I changed the code to use that with just the slight addition of `else if/else`.

Appreciate the time you took to think about this to make it less complex.